Source/JavaScriptCore/runtime/JSArray.cpp

```diff
@@ -1502,4 +1502,96 @@ JSArray* constructArrayNegativeIndexed(JSGlobalObject* globalObject, Structure*
     return array;
 }
 
+template<>
+void clearElement(double& element)
+{
+    element = PNaN;
+}
+
+std::optional<JSArray*> cloneArrayFromFast(JSGlobalObject* globalObject, JSValue arrayValue)
```

Comment:
Let's rename it to `tryCloneArrayFromFast`, and return `nullptr` when it failed.